File: exams/frmEditSplitSubject.vb
Public Class frmEditSplitSubject

    Private Sub frmAddSplitSubject_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Not connect() Then
            Me.Close()
        Else
            load_class(cboClass)
            'todo commented the below code not sure what the repurcsion are
            ' qwrite("ALTER TABLE `split_subjects` ADD COLUMN `weighted` DOUBLE NOT NULL AFTER `contribution`;")
            qread("SELECT * FROM split_subjects WHERE id=" & t_id)
            dbreader.Read()
            Dim subject As String = dbreader("subject")
            names = dbreader("Abbreviation")
            txtAbbreviation.Text = dbreader("Abbreviation")
            txtContribution.Text = dbreader("Contribution")
            txtWeighted.Text = dbreader("weighted")
            txtName.Text = dbreader("Name")
            cboClass.SelectedItem = dbreader("Class")
            cboSubject.SelectedItem = subject
        End If
    End Sub
    Dim names As String = Nothing
    Private Sub load_subjects()
        cboSubject.Items.Clear()
        get_class_subjects(cboClass.SelectedItem)
        For k As Integer = 0 To subjabb.Length - 1
            cboSubject.Items.Add(subjects(k))
        Next
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If cboSubject.SelectedItem <> Nothing And txtName.Text.Trim <> Nothing And txtAbbreviation.Text.Trim <> Nothing And IsNumeric(txtContribution.Text) Then
            start()
            If Not qwrite("UPDATE split_subjects SET class='" & escape_string(cboClass.SelectedItem) & "', subject='" & escape_string(cboSubject.SelectedItem) & "',name='" & escape_string(txtName.Text) &
                          "',abbreviation='" & escape_string(txtAbbreviation.Text) & "', contribution='" & txtContribution.Text & "', weighted='" & txtWeighted.Text & "' WHERE id=" & t_id) Or Not qwrite("ALTER TABLE `exam_split_subject_results`" &
                            "CHANGE `" & get_name(cboClass.SelectedItem) & "_" & names & "` `" & get_name(cboClass.SelectedItem) & "_" & txtAbbreviation.Text & "` VARCHAR(255) NOT NULL DEFAULT '';") Then
                rollback()
                failure("Could Not Save Record!")
            Else
                commit()
                success("Record Successfully Saved!")
                Me.Close()
            End If
        Else
            failure("Please fill in the form correctly!")
        End If
    End Sub

    Private Sub cboClass_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cboClass.SelectedIndexChanged
        load_subjects()
    End Sub

    Private Sub btnDelete_Click(sender As Object, e As EventArgs) Handles btnDelete.Click
        If cboSubject.SelectedItem <> Nothing And txtName.Text.Trim <> Nothing And txtAbbreviation.Text.Trim Then
            If displayConfirmationMessage("Are You Sure You Want To Delete This Subject ?") Then
                If qwrite("ALTER TABLE exam_split_subject_results DROP COLUMN " + txtName.Text + ";") And qwrite("delete from split_subjects where subject = '" + cboSubject.SelectedItem.ToString() + "' and name = '" + txtName.Text + "' and abbreviation = '" + txtAbbreviation.Text + "' ") Then
                    MsgBox("The Operation Was Successful")
                End If
            End If
        Else
            failure("Please fill in the form correctly!")
        End If
    End Sub
End Class

File: exams/frmEditSplitSubject.cs
using System;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace exams
{
    public partial class frmEditSplitSubject
    {
        public frmEditSplitSubject()
        {
            InitializeComponent();
            _Button1.Name = "Button1";
            _cboClass.Name = "cboClass";
            _btnDelete.Name = "btnDelete";
        }

        private void frmAddSplitSubject_Load(object sender, EventArgs e)
        {
            if (!publicSubsNFunctions.connect())
            {
                Close();
            }
            else
            {
                var argcbo = cboClass;
                publicSubsNFunctions.load_class(ref argcbo);
                cboClass = argcbo;
                // todo commented the below code not sure what the repurcsion are
                // qwrite("ALTER TABLE `split_subjects` ADD COLUMN `weighted` DOUBLE NOT NULL AFTER `contribution`;")
                string argq = "SELECT * FROM split_subjects WHERE id=" + publicSubsNFunctions.t_id;
                publicSubsNFunctions.qread(ref argq);
                publicSubsNFunctions.dbreader.Read();
                string subject = Conversions.ToString(publicSubsNFunctions.dbreader["subject"]);
                names = Conversions.ToString(publicSubsNFunctions.dbreader["Abbreviation"]);
                txtAbbreviation.Text = Conversions.ToString(publicSubsNFunctions.dbreader["Abbreviation"]);
                txtContribution.Text = Conversions.ToString(publicSubsNFunctions.dbreader["Contribution"]);
                txtWeighted.Text = Conversions.ToString(publicSubsNFunctions.dbreader["weighted"]);
                txtName.Text = Conversions.ToString(publicSubsNFunctions.dbreader["Name"]);
                cboClass.SelectedItem = publicSubsNFunctions.dbreader["Class"];
                cboSubject.SelectedItem = subject;
            }
        }

        private string names = null;

        private void load_subjects()
        {
            cboSubject.Items.Clear();
            publicSubsNFunctions.get_class_subjects(Conversions.ToString(cboClass.SelectedItem));
            for (int k = 0, loopTo = publicSubsNFunctions.subjabb.Length - 1; k <= loopTo; k++)
                cboSubject.Items.Add(publicSubsNFunctions.subjects[k]);
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (Conversions.ToBoolean(Operators.AndObject(Operators.AndObject(Operators.AndObject(Operators.ConditionalCompareObjectNotEqual(cboSubject.SelectedItem, null, false), !string.IsNullOrEmpty(txtName.Text.Trim())), !string.IsNullOrEmpty(txtAbbreviation.Text.Trim())), Information.IsNumeric(txtContribution.Text))))
            {
                publicSubsNFunctions.start();
                if (!publicSubsNFunctions.qwrite("UPDATE split_subjects SET class='" + publicSubsNFunctions.escape_string(Conversions.ToString(cboClass.SelectedItem)) + "', subject='" + publicSubsNFunctions.escape_string(Conversions.ToString(cboSubject.SelectedItem)) + "',name='" + publicSubsNFunctions.escape_string(txtName.Text) + "',abbreviation='" + publicSubsNFunctions.escape_string(txtAbbreviation.Text) + "', contribution='" + txtContribution.Text + "', weighted='" + txtWeighted.Text + "' WHERE id=" + publicSubsNFunctions.t_id) | !publicSubsNFunctions.qwrite(Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject("ALTER TABLE `exam_split_subject_results`" + "CHANGE `", publicSubsNFunctions.get_name(Conversions.ToString(cboClass.SelectedItem))), "_"), names), "` `"), publicSubsNFunctions.get_name(Conversions.ToString(cboClass.SelectedItem))), "_"), txtAbbreviation.Text), "` VARCHAR(255) NOT NULL DEFAULT '';"))))
                {
                    publicSubsNFunctions.rollback();
                    publicSubsNFunctions.failure("Could Not Save Record!");
                }
                else
                {
                    publicSubsNFunctions.commit();
                    publicSubsNFunctions.success("Record Successfully Saved!");
                    Close();
                }
            }
            else
            {
                publicSubsNFunctions.failure("Please fill in the form correctly!");
            }
        }

        private void cboClass_SelectedIndexChanged(object sender, EventArgs e)
        {
            load_subjects();
        }

        private void btnDelete_Click(object sender, EventArgs e)
        {
            if (Conversions.ToBoolean(Operators.AndObject(Operators.AndObject(Operators.ConditionalCompareObjectNotEqual(cboSubject.SelectedItem, null, false), !string.IsNullOrEmpty(txtName.Text.Trim())), txtAbbreviation.Text.Trim())))
            {
                if (publicSubsNFunctions.displayConfirmationMessage("Are You Sure You Want To Delete This Subject ?"))
                {
                    if (publicSubsNFunctions.qwrite("ALTER TABLE exam_split_subject_results DROP COLUMN " + txtName.Text + ";") & publicSubsNFunctions.qwrite("delete from split_subjects where subject = '" + cboSubject.SelectedItem.ToString() + "' and name = '" + txtName.Text + "' and abbreviation = '" + txtAbbreviation.Text + "' "))
                    {
                        Interaction.MsgBox("The Operation Was Successful");
                    }
                }
            }
            else
            {
                publicSubsNFunctions.failure("Please fill in the form correctly!");
            }
        }
    }
}